mingw-w64-qt5-pkgs/PKGBUILD

```
@@ -282,14 +285,22 @@ package_qt5-base() {
 
   cd ${MSYSTEM}/${_component/5-/}
 
-  make install
-  cd ${pkgdir}/..
-  rm -r ${MINGW_PACKAGE_PREFIX}-${_component}
-  mv ${MINGW_PACKAGE_PREFIX}-qt5 ${MINGW_PACKAGE_PREFIX}-${_component}
+  # Fix Makefiles
+  local PREFIX_WIN=$(cygpath -am ${MINGW_PREFIX})
+  find "${srcdir}/${MSYSTEM}/${_component/5-/}" -type f \( -name 'Makefile*' \) \
+      -exec sed -i -e "s|${PREFIX_WIN:0:2}\$(INSTALL_ROOT|\$(INSTALL_ROOT|g" {} \;
+  find "${srcdir}/${MSYSTEM}/${_component/5-/}" -type f \( -name 'Makefile*' \) \
+      -exec sed -i -e "s|)${PREFIX_WIN:2}|)${MINGW_PREFIX}|g" {} \;
+
+  make INSTALL_ROOT="${pkgdir}" install
 
   install -d "$pkgdir${MINGW_PREFIX}"/share/licenses/${_component}
   install -Dm644 $srcdir/${MSYSTEM}/${_component/5-/}/LICENSE* -t "$pkgdir${MINGW_PREFIX}"/share/licenses/${_component}
 
+  # Seperate debug-info files
+  mv -f "${pkgdir}${MINGW_PREFIX}/lib"/*.debug "${pkgdir}${MINGW_PREFIX}/bin"/ || true
+  rsync -armR --remove-source-files --include="*/" --include="*.debug" --exclude="*" --prune-empty-dirs ${pkgdir}/.${MINGW_PREFIX} ${srcdir}/${MSYSTEM}-${_component/5-/}-debug/
```

```suggestion
  rsync -armR --remove-source-files --include="*/" --include="*.debug" --exclude="*" --prune-empty-dirs ${pkgdir}/${MINGW_PREFIX} ${srcdir}/${MSYSTEM}-${_component/5-/}-debug/
```

?  And several other similar copy/pastes

---

That point is necessary.

---

oh, I see, I forgot `${MINGW_PREFIX}` contains the leading slash

---

You were right, I don't have to put `/.` as `${MINGW_PREFIX}` contains a leading slash.

---

I don't know if rsync has special handling of putting `./` in a path.  I was thinking you would end up with `.mingw64/` for example

---

the manpage says with `-R` a `./` in a path *does* change rsync's behavior... it will create the relative directories after the `./` in the target, rather than creating all of the directories to the root of the source inside the target directory.

---

The first build for this just finished for clang32 with debug packages looking like:
```
2021-08-31T05:28:41.6497717Z [36m[MSYS2 CI](B[m mingw-w64-qt5-pkgs: File listing diff for mingw-w64-clang-i686-qt5-script-debug-5.15.2+kde+r96-1-any.pkg.tar.zst
2021-08-31T05:28:41.6499175Z 
2021-08-31T05:28:41.7898553Z warning: database file for 'ci' does not exist (use '-Fy' to download)
2021-08-31T05:28:47.7101759Z error: package 'mingw-w64-clang-i686-qt5-script-debug' was not found
2021-08-31T05:28:48.2069635Z --- /dev/fd/63	2021-08-31 05:28:41.000000000 +0000
2021-08-31T05:28:48.2106016Z +++ /dev/fd/62	2021-08-31 05:28:41.000000000 +0000
2021-08-31T05:28:48.2116245Z @@ -0,0 +1,5 @@
2021-08-31T05:28:48.2125792Z +/CLANG32-qtscript-debug/
2021-08-31T05:28:48.2126592Z +/CLANG32-qtscript-debug/clang32/
2021-08-31T05:28:48.2127355Z +/CLANG32-qtscript-debug/clang32/bin/
2021-08-31T05:28:48.2128542Z +/CLANG32-qtscript-debug/clang32/bin/Qt5Script.dll.debug
2021-08-31T05:28:48.2129654Z +/CLANG32-qtscript-debug/clang32/bin/Qt5ScriptTools.dll.debug
2021-08-31T05:28:48.2131588Z ##[endgroup]
```

I expect even deeper paths in the latest build.  I recommend checking the diff sections for the new debug packages to double-check the paths are right before merging.